Userland/Libraries/LibGfx/ImageFormats/TinyVGLoader.cpp

```diff
@@ -127,10 +127,29 @@ static ErrorOr<TinyVGHeader> decode_tinyvg_header(Stream& stream)
     return header;
 }
 
-static ErrorOr<FixedArray<Color>> decode_color_table(Stream& stream, ColorEncoding encoding, u32 color_count)
+static ErrorOr<FixedArray<Color>> decode_color_table(FixedMemoryStream& stream, ColorEncoding encoding, u32 color_count)
 {
     if (encoding == ColorEncoding::Custom)
         return Error::from_string_literal("Invalid TinyVG: Unsupported color encoding");
+
+    auto expected_color_table_size = [](u32 color_count, ColorEncoding color_encoding) -> u64 {
+        switch (color_encoding) {
+        case ColorEncoding::RGBA8888:
+            return color_count * 4;
+        case ColorEncoding::RGB565:
+            return color_count * 2;
+        case ColorEncoding::RGBAF32:
+            return color_count * 16;
+        case ColorEncoding::Custom:
+            break;
+        }
+
+        VERIFY_NOT_REACHED();
+    };
+
+    if (stream.remaining() <= expected_color_table_size(color_count, encoding))
+        return Error::from_string_literal("Invalid TinyVG: Reported color table size larger than the remaining number of bytes");
+
```

Comment:
You can keep the API general to any stream type, and just check if the stream is a `FixedMemoryStream` (and do this check in that case):

```suggestion
     if (is<FixedMemoryStream>(stream)) {
       auto& fixed_stream = static_cast<FixedMemoryStream&>(stream);
       if (fixed_stream.remaining() <= expected_color_table_size(color_count, encoding))
         return Error::from_string_literal("Invalid TinyVG: Reported color table size larger than the remaining number of bytes");
     }
```

Reply:
(more generally you could do the check via `SeekableStream`, though the API is a little more painful)